File: src/main/java/br/com/rotaractsorocabaleste/rotasales/saleitem/SaleItemServiceImpl.java
package br.com.rotaractsorocabaleste.rotasales.saleitem;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.HashSet;
import java.util.Set;
import java.util.UUID;

@Service
@RequiredArgsConstructor
public class SaleItemServiceImpl implements SaleItemService {

    private final SaleItemRepository saleItemRepository;

    @Override
    public Set<SaleItem> saveAll(final Set<SaleItem> items) {
        return new HashSet<>(saleItemRepository.saveAll(items));
    }

    @Override
    public void deleteAllByIdInBatch(Set<UUID> ids) {
        saleItemRepository.deleteAllByIdInBatch(ids);
    }

}

File: src/main/kotlin/br/com/rotaractsorocabaleste/rotasales/core/service/impl/SaleItemServiceImpl.kt
package br.com.rotaractsorocabaleste.rotasales.core.service.impl

import br.com.rotaractsorocabaleste.rotasales.core.entity.SaleItem
import br.com.rotaractsorocabaleste.rotasales.core.service.SaleItemService
import br.com.rotaractsorocabaleste.rotasales.dataprovider.repository.SaleItemRepository
import org.springframework.stereotype.Service
import java.util.*

@Service
class SaleItemServiceImpl(
    private val saleItemRepository: SaleItemRepository
) : SaleItemService {
    override fun saveSaleItems(saleItems: List<SaleItem>): List<SaleItem> =
        saleItemRepository
            .saveAll(saleItems)
            .toMutableList()

    override fun getSaleItemsBySaleId(saleId: UUID): List<SaleItem> =
        saleItemRepository.findBySaleId(saleId)

    override fun removeSaleItems(saleItems: List<SaleItem>) =
        saleItemRepository.deleteAll(saleItems)

}